WiinUPro/Windows/DevicePrefsWindow.xaml.cs

```diff
@@ -23,23 +23,27 @@ public partial class DevicePrefsWindow : Window
         
         private DevicePrefs _modifiedPrefs;
 
+        private int lastIndex;
+        private bool selectionChanged = false;
+
         protected DevicePrefsWindow()
         {
             InitializeComponent();
         }
 
-        public DevicePrefsWindow(DevicePrefs devicePrefs) : this()
+        public DevicePrefsWindow(DevicePrefs devicePrefs, string type = null) : this()

```

Comment:
I believe I understand why type is a string here, but the string comparison stuff below seems like a non-ideal design, but I think it's fine for now. There's plenty of stuff in the code that should probably be approached differently.

Reply:
Right, it doesn't have to be of string type anymore. My original code required type to be a string (I compared `type` with the text in the ComboBox instead of comparing with the `extTypes` list at line 93, an approach which had several drawbacks), but there's no reason for it to be of string type now. I'll change it to ControllerType instead.